documentation/quarkus/guide-beanmapping-quarkus.asciidoc

```diff
@@ -0,0 +1,224 @@
+    </configuration>
+</plugin>
+----
+In our https://github.com/devonfw-sample/devon4quarkus-reference[quarkus reference project] you can get a look into the usage of both frameworks.
+
+== MapStruct Configuration
+
+We already discussed the benefits from https://github.com/devonfw/devon4j/blob/master/documentation/guide-dependency-injection.asciidoc#dependency-injection[dependency injection] and MapStruct supports CDI with EJB, spring, and jsr330. The default retrieving method for a mapper is a factory that uses reflections and should be avoided. The component model should be set to CDI, as this will allow us to easily inject the generated mapper implementation. The component model can be configured in multiple ways.
+
+=== Simple Configuration
+Add the attribute *componentModel* to the *@Mapper* annotation in the mapper interface.
+[source, java]
+----
+@Mapper(compnentModel = "cdi")
+public interface AnimalMapper{
+  ... 
+}
+----
+
+=== MapperConfig Configuration
+Create a shared configuration that can be used for multiple mappers. Implement an Interface and use the annotation *@MapperConfig* for the class. You can define all configurations in this interface and pass the generated `MapperConfig.class` with the *config* attribute to the mapper. The `MapperConfig` also defines the *InjectionStrategy* and *MappingInheritaceStrategy* both will be explained later.
+A list of all configurations can be found https://mapstruct.org/documentation/stable/api/org/mapstruct/MapperConfig.html[here].
+[source, java]
+----
+@MapperConfig(
+  compnentModel = "cdi",
+  mappingInheritanceStrategy = MappingInheritanceStrategy.AUTO_INHERIT_FROM_CONFIG
+  injectionStrategy =InjectionStrategy.CONSTRUCTOR
+)
+public interface MapperConfig{
+}
+----
+
+[source, java]
+----
+@Mapper( config = MapperConfig.class )
+public interface AnimalMapper{
+  ...
+}
+----
+Any attributes notgiven via *@Mapper* will be inherited from the shared configuration `MapperConfig.class`.
+
+=== Configuration via annotaion processor options
+The MapStruct code generator can be configured using annotation processor options.
+You can pass the options to the compiler while invoking javac directly, or add the parameters to the maven configuration in the POM.xml
+[source, xml]
+----
+<build>
+  <plugins>
+    <plugin>
+      <artifactId>maven-compiler-plugin</artifactId>
+      <version>${compiler-plugin.version}</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
+          <annotationProcessorPaths>
+            <path>
+                <groupId>org.mapstruct</groupId>
+                <artifactId>mapstruct-processor</artifactId>
+                <version>${org.mapstruct.version}</version>
+            </path>
+          </annotationProcessorPaths>
+        <parameters>true</parameters>
+        <compilerArgs>
+          <!-- mapstruct, please generate all classes as CDI beans -->
+          <arg>-Amapstruct.defaultComponentModel=cdi</arg>
```

Comment:
You show three different ways to do the same configuration. I think it would be sufficient to show configurations via `@Mapper` and `@MapperConfig` and just mention that defaults can be set also in the pom and add some link to the corresponding part in the official guide